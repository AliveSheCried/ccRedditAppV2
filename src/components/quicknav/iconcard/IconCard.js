import React from "react";
import { svgHot, svgNew, svgRising, svgTop } from "../../../assets/images/svg";

export const IconCard = ({ svg, label, endpoint }) => {
  let selectedSvg;
  if (svg === "svgHot") {
    selectedSvg = svgHot;
  } else if (svg === "svgNew") {
    selectedSvg = svgNew;
  } else if (svg === "svgTop") {
    selectedSvg = svgTop;
  } else if (svg === "svgRising") {
    selectedSvg = svgRising;
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    let selectedEndpoint = event.currentTarget.dataset.endpoint;
    console.log(selectedEndpoint);
  };

  return (
    <button
      className="icons__btn"
      type="submit"
      onClick={handleSubmit}
      data-endpoint={endpoint}
    >
      {selectedSvg}
      <span className="icons__btn-label">{label}</span>
    </button>
  );
};
