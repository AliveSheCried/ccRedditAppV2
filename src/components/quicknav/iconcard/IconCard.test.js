import React from "react";
import { shallow } from "enzyme";
import { IconCard } from "./IconCard";

describe("IconCard component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<IconCard />);
  });

  describe("Initial rendering", () => {
    it("should match the snapshot", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it("should include a button", () => {
      expect(wrapper.find("button").length).toEqual(1);
    });

    it("should include correct props for button 1/4", () => {
      expect(wrapper.find("button").prop("className")).toEqual("icons__btn");
    });

    it("should include correct props for button 2/4", () => {
      expect(wrapper.find("button").prop("type")).toEqual("submit");
    });

    it("should include correct props for button 3/4", () => {
      expect(wrapper.find("button").prop("onClick")).toEqual(
        expect.any(Function)
      );
    });

    it("should include correct props for button 4/4", () => {
      expect(wrapper.find("button").prop("data-endpoint")).toEqual(undefined);
    });
  });

  /// not able to get this to work - raise help request at stackoverflow.
//   describe("functionality of IconCard", () => {
//     it("should call handleSubmit", () => {
//       console.log(wrapper.debug());
//       const event = {
//         preventDefault() {},
//         currentTarget: { dataset: "hot" },
//       };
//       const mockFn = jest.fn();
//       IconCard.handleSubmit = function() {
//         mockFn();
//       };
//       wrapper.find("button").simulate("click", event);
//       expect(mockFn).toHaveBeenCalledTimes(1);
//     });
//   });
// });

// describe("IconCard component will render & function correctly", () => {
//   let wrapper;

//   beforeEach(() => {
//     wrapper = shallow(<IconCard />);
//   });

//   test("should render single wrapper div with specific class", () => {
//     expect(wrapper.find(".icons__btn")).toHaveLength(1);
//   });

//   test("should render hot IconCard when label prop is 'hot'", () => {
//     expect(wrapper.exists(".icons__btn-label")).toEqual(true);
//     wrapper.unmount();
//   });

//   test("handleSubmit is called", () => {
//     const logSpy = jest.spyOn(console, "log");
//     const event = {
//       preventDefault() {},
//       currentTarget: { dataset: "hot" },
//     };
//     //const wrapper = shallow(<SampleComponent></SampleComponent>);
//     const button = wrapper.find("button");
//     //expect(button.text()).toBe('Click Me');
//     button.simulate("click", event);
//     expect(logSpy).toBeCalledWith("hot");
//   });

// test("when button is click, correct endpoint is sent to handle function", () => {
//   const mockHandleSubmit = jest.fn();
//   const event = {
//     preventDefault() {},
//     currentTarget: { dataset: "hot" },
//   };
//   const component = shallow(<IconCard onClick={mockHandleSubmit} />);
//   component.find("button").simulate("click", event);
//   expect(mockHandleSubmit).toBeCalledWith("hot");
// });
//});
