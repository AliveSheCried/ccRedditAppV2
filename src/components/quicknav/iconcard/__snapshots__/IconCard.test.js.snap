// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconCard component Initial rendering should match the snapshot 1`] = `"<button class=\\"icons__btn\\" type=\\"submit\\"><span class=\\"icons__btn-label\\"></span></button>"`;
